timeless/customers/models.py

```diff
@@ -22,24 +22,34 @@ def __repr__(self):
         return "<Customer(name=%s %s)>" % (self.first_name, self.last_name)
 
     @classmethod
-    def merge_with_poster(cls, customer, poster_customer: dict):
+    def merge_with_poster(cls, customer: "Customer", poster_customer: dict):
         """
         Method should return Customer object with merged data from table entity
         and poster customer dict
-        @todo #262:30min Implement two class methods merge_with_poster and
-         create_by_poster. merge_with_poster will merge entry entity with
-         poster entity, we should make right fields mapping, as result
-         returns Customer instance.
-         The same should be made with method create_by_poster, returns Customer
-         instance with data from poster_customer
         """
-        return cls()
+        return Customer(
+            id=customer.id,
+            first_name=customer.first_name,
+            last_name=customer.last_name,
+            phone_number=poster_customer["phone_number"],
+            created_on=customer.created_on,
+            updated_on=datetime.utcnow(),
+            poster_id=poster_customer["client_id"],
+            synchronized_on=datetime.utcnow()
```

Comment:
@soin08 How do you know which fields should be set from Poster and which from the given ``customer``?

Reply:
@amihaiemil if we look at how these methods are used in /timeless/poster/tasks.py 
```def sync_customers():
    """
    Periodic task for fetching and saving tables from Poster
    """
    for poster_customer in __poster_api().customers():
        customer = DB.session(Customer).query.filter_by(
            first_name=poster_customer["first_name"],
            last_name=poster_customer["last_name"]
        ).first()
```
we can see that the matching is done by `first_name` and `last_name`. Hence the rest of the fields needs to be merged from poster. `first_name` and `last_name` stay the same so it doesn't matter whether to take them from poster or from customer.

Reply:
@soin08 I see, thanks for explaining :)

Reply:
@soin08 actually this is not correct, however this is not your fault but mine: the matching should be done thru poster_id attribute stored in customer. Please check this is the way it works, otherwise please add a puzzle to fix it thanks

Reply:
@emilianodellacasa Since the merging is done by poster_id, I added `first_name` and `last_name` to be merged from poster. It's good to go.